OS X test fixes

adamtheturtle opened the pull request:
These are changes to Flocker to get the test suite to pass on OS X. We only care about the CLI working so any skips which are not in the CLI are fine.

See tests passing / skipping on OS X at http://build.staging.clusterhq.com/builders/flocker-osx-10.10/builds/3.

tomprince commented on flocker/node/testtools.py:
```diff
@@ -4,14 +4,26 @@
+    """
+    :return: ``True`` if Docker is installed and running, else ``False``.
+
+    XXX This is terrible (https://clusterhq.atlassian.net/browse/FLOC-85)
```
It would probably be better to use something like https://github.com/ClusterHQ/flocker/commit/6c0dcb8c980c35f7cfb4989b1dbd82fd14a6c778 since we probably *don't* want to use boot2docker in the tests. (Maybe along with https://github.com/ClusterHQ/flocker/blob/master/flocker/node/_docker.py#L271).

tomprince commented on setup.py:
```diff
@@ -35,6 +36,50 @@
+
+# The test suite uses network namespaces
+# nomenclature cannot be installed on OS X
+if platform.system() != 'Darwin':
```
Better to check for `Linux`, since nomenclature can only be installed there.

tomprince commented on flocker/cli/functional/test_sshconfig.py:
```diff
@@ -64,7 +64,8 @@ def test_connection_failed(self):
                                 self.configure_ssh, b"127.0.0.1", port)
-        self.assertIn(b"refused", exc.output)
+        self.assertTrue(b"refused" in exc.output or "Operation timed out" in
+                        exc.output)
```
I guess a bound socket that isn't listening behaves differently on OSX. I guess there should be a comment explaining the possible behaviors.

tomprince commented:
Please address the above comments. I'm rejecting it to review the docker skip changes.

adamtheturtle commented:
I have addressed the review comments, making minor changes to the linked previously-existing code. If this is suitable, check if https://clusterhq.atlassian.net/browse/FLOC-85 should be marked fixed too.

tomprince commented on fabfile.py:
```diff
@@ -0,0 +1,8 @@
+from fabric.api import task, run, local
+
+@task
+def brew():
```
This file doesn't belong in the root like this.

tomprince commented on flocker/cli/functional/test_sshconfig.py:
```diff
@@ -64,7 +64,13 @@ def test_connection_failed(self):
         exc = self.assertRaises(CalledProcessError,
                                 self.configure_ssh, b"127.0.0.1", port)
-        self.assertIn(b"refused", exc.output)
+        # There are different error messages on different platforms.
```
It'd be nice if there was some comment on *why* there are different error messages.

adamtheturtle replied:
A quick search shows nothing and I don't know - I'm not going to dig into this although I agree that it would be nice.

tomprince commented on flocker/node/testtools.py:
```diff
@@ -4,14 +4,43 @@
-# This is terible (https://clusterhq.atlassian.net/browse/FLOC-85):
-if_docker_configured = skipIf(Popen([b"docker", b"version"]).wait(),
-                              "Docker must be installed and running.")
+SOCKET_PATH = BASE_DOCKER_API_URL.split(':/')[-1]
```
This should probably have `DOCKER` in the name.

tomprince commented on start_homebrew_machine:
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
```
Same with these scripts.

tomprince commented on flocker/node/testtools.py:
```diff
@@ -4,14 +4,43 @@
+    else:
+        return True
+
+if_docker_configured = skipUnless(
```
This now duplicates https://github.com/ClusterHQ/flocker/commit/6c0dcb8c980c35f7cfb4989b1dbd82fd14a6c778, so we should get rid of the other one (and just use this).

adamtheturtle replied:
That code seems to be gone from `master`.

adamtheturtle commented:
Apologies @tomprince , I had WIP that was meant to be in another branch and pushed it here, which is why the fabric stuff is here. I will clean this up.

adamtheturtle commented:
Passing build.

adamtheturtle commented:
http://build.clusterhq.com/builders/flocker-osx-10.10/builds/5
